src/librustc_typeck/check/mod.rs

```diff
@@ -1761,6 +1761,7 @@ pub fn check_simd(tcx: TyCtxt<'_>, sp: Span, def_id: DefId) {
             match e.sty {
                 ty::Param(_) => { /* struct<T>(T, T, T, T) is ok */ }
```

Comment:
Was this added later? It effectively invalidates the checks below...
Even more reason we should have a single `#[lang = "simd"]` instead of many `#[repr(simd)]`.

Reply:
I think this was added at some point last year to support generic vectors.

Reply:
For the time being at least we probably want to type check SIMD vectors early, e.g., `#[repr(simd)] struct Foo(Vec<i32>, Vec<i32>);` and `#[repr(simd)] struct Bar<const N: usize>([Vec<i32>;  N]);` should fail to type check. 

If we move to a single:

```rust
#[lang = "simd"]
struct Simd<T, const N: usize>([T; N]) where [T; N]: SomeTrait;
```

in the future then we might just accept anything and leave rejecting invalid inputs to `SomeTrait`.